src/pages/certificates/create-certificate/create-certificate.js

```diff
@@ -7,37 +10,82 @@ import {
   TextField,
   Tooltip
 } from '@material-ui/core';
-import { useStyles } from './create-certificate.styles';
-import { BackButton } from '../../../components/buttons';
-import materialUiConstants from '../../../configs/material-ui-constants';
-import { useCommonStyles } from '../../common.styles';
-import CheckBoxes from '../checkBoxes';
+
+import { LOCAL_STORAGE } from '../../../consts/local-storage';
+import { getFromLocalStorage } from '../../../services/local-storage.service';
+
+import {
+  setSnackBarMessage,
+  setSnackBarSeverity,
+  setSnackBarStatus
+} from '../../../redux/snackbar/snackbar.actions';
+import LoadingBar from '../../../components/loading-bar';
+
 import formRegExp from '../../../configs/form-regexp';
-import CertificatesTable from '../certificatesTable';
 import { loginErrorMessages } from '../../../configs/error-messages';
 import buttonTitles from '../../../configs/button-titles';
 import titles from '../../../configs/titles';
 import routes from '../../../configs/routes';
+import materialUiConstants from '../../../configs/material-ui-constants';
 
-const { certificatesTitles } = titles;
+import CheckBoxes from '../checkBoxes';
+import { BackButton } from '../../../components/buttons';
+import { useCommonStyles } from '../../common.styles';
+
+import { useStyles } from './create-certificate.styles';
+import CertificatesTable from '../certificatesTable';
+import { bulkGenerateCertificates } from '../operations/certificate.mutation';
+
+const { certificatesTitles, certificatesValueTitles } = titles;
 
 const CreateCertificate = () => {
+  const dispatch = useDispatch();
+
   const commonStyles = useCommonStyles();
   const styles = useStyles();
 
-  const initialCheckboxes = [
-    { checked: false, quantity: 1, name: certificatesTitles[500] },
-    { checked: false, quantity: 1, name: certificatesTitles[1000] },
-    { checked: false, quantity: 1, name: certificatesTitles[1500] }
-  ];
+  const arrValues = Object.keys(certificatesValueTitles);
+  const initialCheckboxes = arrValues.map((key) => ({
+    checked: false,
+    quantity: 1,
+    name: certificatesValueTitles[key],
+    value: Number(key)
+  }));
 
   const [checkBoxes, setCheckBoxes] = useState(initialCheckboxes);
-  const [date, setDate] = useState(new Date());
+  const [date, setDate] = useState();
   const [email, setEmail] = useState('');
   const [isInvalid, setIsInvalid] = useState(false);
   const [certificates, setCertificates] = useState([]);
   const [disabled, setDisabled] = useState(true);
 
+  const token = getFromLocalStorage(LOCAL_STORAGE.AUTH_ACCESS_TOKEN);
+
+  const [generateCertificates, { loading: certificatesLoading }] = useMutation(
+    bulkGenerateCertificates,
+    {
+      context: {
```

Comment:
I will ask you to move context settings with the token, to Appollo Provider general settings.

https://www.apollographql.com/docs/react/api/link/apollo-link-context/

Reply:
the context with the token is already in the general settings, so Andiy can simply delete it from his code

Reply:
> the context with the token is already in the general settings, so Andiy can simply delete it from his code

thanks, Taras

Reply:
done